### PR TITLE
Introduce bundling with esbuild, `--hotswap` and `--watch`

### PR DESCRIPTION
**Pull request by samchungy**
Related: https://github.com/seek-oss/skuba/issues/1311

### CODE REVIEW

File: template/lambda-sqs-worker-cdk/infra/__snapshots__/appStack.test.ts.snap [hunk 1]
```diff
@@ -163,7 +163,7 @@ exports[`returns expected CloudFormation stack for dev 1`] = `
           },
         },
         "FunctionName": "serviceName",
-        "Handler": "app.handler",
+        "Handler": "index.handler",
```
Comment (samchungy) [inline template/lambda-sqs-worker-cdk/infra/__snapshots__/appStack.test.ts.snap:166]: >The handler value will be automatically prefixed with the bundled output file name, index., unless the handler value contains a . character, in which case the handler value is used as-is to allow for values needed by some Lambda extensions.

File: template/lambda-sqs-worker-cdk/infra/appStack.ts [hunk 2]
```diff
@@ -52,11 +53,14 @@ export class AppStack extends Stack {
...
+    const worker = new aws_lambda_nodejs.NodejsFunction(this, 'worker', {
       architecture: aws_lambda.Architecture[architecture],
-      code: new aws_lambda.AssetCode('./lib'),
+      entry: './src/app.ts',
+      bundling: {
+        sourceMap: true,
+        target: 'node20',
+      },
       runtime: aws_lambda.Runtime.NODEJS_20_X,
-      handler: 'app.handler',
```
Comment (samchungy) [inline template/lambda-sqs-worker-cdk/infra/appStack.ts:59]: >Alternatively, an entry file and handler can be specified:

new nodejs.NodejsFunction(this, 'MyFunction', {
  entry: '/path/to/my/file.ts', // accepts .js, .jsx, .cjs, .mjs, .ts, .tsx, .cts and .mts files
  handler: 'myExportedFunc', // defaults to 'handler'
});

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs-readme.html#configuring-esbuild

File: template/lambda-sqs-worker-cdk/infra/__snapshots__/appStack.test.ts.snap (hunk 1 above)
Comment (72636c) [reply template/lambda-sqs-worker-cdk/infra/__snapshots__/appStack.test.ts.snap:166]: I wonder if we should rename the source file to `index.ts` to align 🤔

Comment (samchungy) [reply template/lambda-sqs-worker-cdk/infra/__snapshots__/appStack.test.ts.snap:166]: I don't think it's very necessary, just a bundling thing tbh

File: template/lambda-sqs-worker-cdk/package.json [hunk 3]
```diff
@@ -4,6 +4,8 @@
   "scripts": {
     "build": "skuba build",
```
Comment (72636c) [inline template/lambda-sqs-worker-cdk/package.json:5]: Does this render `skuba build` redundant for this template?

File: template/lambda-sqs-worker-cdk/package.json [hunk 4]
```diff
@@ -4,6 +4,8 @@
...
     "deploy": "cdk deploy appStack --require-approval never --context stage=${ENVIRONMENT}",
+    "deploy:hotswap": "yarn deploy --hotswap",
+    "deploy:watch": "yarn deploy:hotswap --watch",
     "format": "skuba format",
     "lint": "skuba lint",
     "package": "yarn install --ignore-optional --ignore-scripts --modules-folder ./lib/node_modules --non-interactive --offline --production",
```
Comment (72636c) [inline template/lambda-sqs-worker-cdk/package.json:11]: Do we still need this packaging malarkey?

**Review by 72636c — APPROVED** [review]
👏

File: template/lambda-sqs-worker-cdk/package.json (hunk 3 above)
Comment (samchungy) [reply template/lambda-sqs-worker-cdk/package.json]: Ruh roh looks like removing this causes issues with test-template

Comment (72636c) [reply template/lambda-sqs-worker-cdk/package.json]: Heh, I think we can catch and ignore "Command not found" errors there 🙏

**Review by 72636c — APPROVED** [review]

**Pull request merged**